File: ruby/gilded_rose.rb
require_relative 'gilded_rose_helper'

class GildedRose
  def initialize(items)
    @items = items
  end

  def update_quality
    @items.each { |item| GildedRoseHelper.update(item) }
  end
end

class Item
  attr_accessor :name, :sell_in, :quality

  def initialize(name, sell_in, quality)
    @name = name
    @sell_in = sell_in
    @quality = quality
  end

  def to_s
    "#{@name}, #{@sell_in}, #{@quality}"
  end
end

File: js/src/gilded_rose.js
import GildedRoseHelper from "./gilded_rose_helper";

export class Item {
  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}

export class Shop {
  constructor(items = []) {
    this.items = items;
  }
  updateQuality() {
    for (var i = 0; i < this.items.length; i++) {
      GildedRoseHelper.update(this.items[i]);
    }

    return this.items;
  }
}
